### ISSUE TITLE
clean Error:Execution failed for task ':app:mockableAndroidJar'. > java.lang.NullPointerException (no error message) in Android Studio 2.2

### ISSUE THREAD

**Issue by jp1017** [anggrayudi/android-hidden-api#21]
Error: 

![20161004103811](https://cloud.githubusercontent.com/assets/7868514/19060924/de57ae08-8a1e-11e6-98fe-67052792023b.jpg)

gradle:

···
    compileSdkVersion 23
    buildToolsVersion "23.0.3"
···

This error comes when using your lib of android-23, and it disappear when using the original android-23.jar.

We know `task ':app:mockableAndroidJar'` is experimental in Android Studio 2.1.3, and I close this.

[Android Studio clean 时产生 Error:Execution failed for task ':app:mockableAndroidJar' > java.lang.NullPointerException (no error message)](https://jp1017.github.io/2016/06/22/Android-Studio-clean-%E6%97%B6%E4%BA%A7%E7%94%9F-Error-Execution-failed-for-task-app-mockableAndroidJar%20java.lang.NullPointerException%20(no%20error%20message)/)

 but google makes this stable in Android Stuido 2.2, and I can't this.

This error comes for some days until today I found this may be result from your lib.

**Comment by anggrayudi** [comment]
I also still don't know what happens with hidden API v23. On some users, this error doesn't appear. Otherwise, some users are facing this problem. For me, I don't face this problem.

**Comment by benniaobuguai** [comment]
I also met this problem today. Are you have solution now ?

**Comment by KaelLuo** [comment]
I met this problem days ago, for now I clean all the files by hand every every time...., and this do makes me annoy. Does anyone find the solution about this ?

**Comment by IgorEisberg** [comment]
I have this issue as well as I'm using a modified android.jar (which has all the hidden API revealed).
I don't use the unit testing and there's no apparent way to disable it in Android Studio 2.2.2, so I found a workaround for this problem:
1) Go to File -> Settings -> Build, Execution, Deployment -> Compiler
2) Add to "Command-line Options":
`-x :app:prepareReleaseUnitTestDependencies -x :app:mockableAndroidJar -x :app:compileReleaseUnitTestSources`
3) Press "OK" and try to Rebuild Project again.

Now the unit testing compilation tasks will always be skipped and no errors will be thrown.

**Comment by jp1017** [comment]
@IgorEisberg wow, thanks, :+1: :smile:  it works. I just using 

>-x :app:mockableAndroidJar

**Comment by IgorEisberg** [comment]
@jp1017 That works too, I just feel as if disabling all of the "unit testing" tasks is better because it's also faster (you have 3 less tasks instead of just 1 task less).
Enjoy!

**Comment by jp1017** [comment]
Yes, but I can't disable others for I using [Tinker](https://github.com/Tencent/tinker) which using thems. :disappointed:

**Comment by IgorEisberg** [comment]
@jp1017 Oh fair enough. Well, at least the problem's solved for now! Gradle is such a buggy thing... -_-

**Comment by tchgithub** [comment]
@IgorEisberg  I add  -x :app:prepareReleaseUnitTestDependencies -x :app:mockableAndroidJar -x :app:compileReleaseUnitTestSources to my settings   but it still show Error:java.lang.NullPointerException (no error message)     
Information:Gradle tasks [:api:generateDebugSources, :api:generateDebugAndroidTestSources, :api:mockableAndroidJar, :api:prepareDebugUnitTestDependencies, :app:generateDebugSources, :app:generateDebugAndroidTestSources, :app:mockableAndroidJar, :app:prepareDebugUnitTestDependencies, :core:generateDebugSources, :core:mockableAndroidJar, :core:prepareDebugUnitTestDependencies, :core:generateDebugAndroidTestSources, :library:generateDebugSources, :library:generateDebugAndroidTestSources, :library:mockableAndroidJar, :library:prepareDebugUnitTestDependencies, :model:generateDebugSources, :model:mockableAndroidJar, :model:prepareDebugUnitTestDependencies, :model:generateDebugAndroidTestSources]

**Comment by jp1017** [comment]
@tchgithub 
> :api:mockableAndroidJar,

For this, you should add below to your setting: 

>-x :api:mockableAndroidJar

For others, do it like this. enjoy it. :smile:

**Comment by IgorEisberg** [comment]
Another workaround: In your project's main build.gradle, inside "allprojects":
`allprojects {
    repositories {
        jcenter()
    }
    gradle.taskGraph.whenReady {
        tasks.each { task ->
            if (task.name.equals('mockableAndroidJar')) {
                task.enabled = false
            }
        }
    }
}`

**Comment by IgorEisberg** [comment]
Another workaround: In your project's main build.gradle, inside "allprojects":
```
allprojects {
    repositories {
        jcenter()
    }
    gradle.taskGraph.whenReady {
        tasks.each { task ->
            if (task.name.equals('mockableAndroidJar')) {
                task.enabled = false
            }
        }
    }
}
```

**Comment by jp1017** [comment]
> Another workaround: In your project's main build.gradle, inside "allprojects":

@IgorEisberg :+1: After add this to my build.gradle, I found this in `gradle console`: 

```
:app:mockableAndroidJar SKIPPED
```

It worked too. :+1: , Thank you.

**Comment by IgorEisberg** [comment]
Yup, it covers the whole project, no matter how you name your modules. It's also more portable since it'll work immediately even if you transfer the project to another computer.
To skip all Test tasks you can change the "if" statement to:
`if (task.name.contains('Test') || task.name.equals('mockableAndroidJar')) {`

**Comment by hyy007** [comment]
I do like this, but it didn't work. Please can you give another way to resolve it? Thx a lot! @jp1017

**Comment by jp1017** [comment]
@hyy007 May be you should show some codes.

**Comment by hyy007** [comment]
java.lang.NullPointerException(no error message). That's the error yestoday, and then when I rebuild the project, it shows: " Error:org.gradle.api.internal.changedetection.state.FileCollectionSnapshotImpl cannot be cast to org.gradle.api.internal.changedetection.state.OutputFilesCollectionSnapshotter$OutputFilesSnapshot
Possible causes for this unexpected error include:<ul><li>Gradle's dependency cache may be corrupt (this sometimes occurs after a network connection timeout.)
<a href="syncProject">Re-download dependencies and sync project (requires network)</a></li><li>The state of a Gradle build process (daemon) may be corrupt. Stopping all Gradle daemons may solve this problem.
<a href="stopGradleDaemons">Stop Gradle build processes (requires restart)</a></li><li>Your project may be using a third-party plugin which is not compatible with the other plugins in the project or the version of Gradle requested by the project.</li></ul>In the case of corrupt Gradle processes, you can also try closing the IDE and then killing all Java processes."
But I don't know the where's the right question... I restarted the Android Studio or restart the PC (Windows 7 system), it didn't help, either~ ‘(*>﹏<*)′ wuwu~ @jp1017

**Comment by jp1017** [comment]
@hyy007 what for other projects? gradle version? just try any you can.

**Comment by hyy007** [comment]
Thank you for your answers for me. I checked out a new version and reopened it, then it went OK just like before... And I still don't know what's the question....I'm not sure if it is the SVN's problem when I work with my teammates.

**Comment by anggrayudi** [comment]
Try Hidden API v24 and this problem will be solved.

**Comment by jp1017** [comment]
@anggrayudi  25 still existed while the first compiling.

**Comment by xicomdev** [comment]
Error:Execution failed for task ':app:processDebugManifest'.
> Manifest merger failed with multiple errors, see logs

dependencies {
    compile fileTree(include: ['*.jar'], dir: 'libs')
    androidTestCompile('com.android.support.test.espresso:espresso-core:2.2.2', {
        exclude group: 'com.android.support', module: 'support-annotations'
    })



    compile 'com.victor:lib:1.0.4'
    compile 'com.android.support:multidex:1.0.1'
    compile 'com.squareup.retrofit:retrofit:1.9.0'
    compile 'com.jeremyfeinstein.slidingmenu:library:1.3@aar'
    compile 'com.android.support:appcompat-v7:25.0.1'
    compile 'com.android.support:design:25.0.1'
    compile 'com.android.support:recyclerview-v7:25.0.1'
    compile 'com.android.support:cardview-v7:25.0.1'
    compile 'de.hdodenhof:circleimageview:2.1.0'

    compile 'uk.co.chrisjenx:calligraphy:2.2.0'
    compile 'com.jakewharton:butterknife:8.4.0'
    annotationProcessor 'com.jakewharton:butterknife-compiler:8.4.0'
    testCompile 'junit:junit:4.12'
    compile 'com.braintreepayments.api:drop-in:3.+'
    compile 'com.github.esafirm.android-image-picker:imagepicker:1.3.2'
    compile 'com.github.bumptech.glide:glide:3.7.0'

    compile 'com.google.android.gms:play-services:9.8.0'
    compile 'com.android.support:support-v4:25.0.1'


    compile 'com.opentok.android:opentok-android-sdk:2.10.+'
    compile 'com.google.firebase:firebase-core:9.8.0'
    compile 'com.google.firebase:firebase-messaging:9.8.0'
    compile 'com.google.code.gson:gson:2.+'

    compile(name:'instaply-sdk-1.0.2', ext:'aar')

}
apply plugin: 'com.google.gms.google-services'

Any suggestion

**Comment by AndroidNinza** [comment]
I have faced same issue and i have tried above all the answers but it want work for me. I think its worked in other cases. After 2 to 3 hours i will got the solution i am just extending the answer so i post this comment.  In case my current project `minimumSdk 15` and my modal project or dependency library's `minimumSdk 18` so i have change my current project `minimumSdk 18` and it worked for me. So check your library project's version

**Comment by cikal** [comment]
@IgorEisberg wow thanks dude.. :+1:
just try your trick and my problem get solved, you save my day..

>    `-x :app:prepareReleaseUnitTestDependencies -x :app:mockableAndroidJar -x :app:compileReleaseUnitTestSources`

**Comment by IgorEisberg** [comment]
@cikal Glad it's still useful!

**Comment by anggrayudi** [comment]
Closing as obsolete — this targets a long-superseded toolchain (Android Studio 2.2, from 2016). The project has since moved to AGP 9 / SDK 37, and the prebuilt Google Drive jars are deprecated in favor of the **`hiddenjar`** CLI, which builds a fresh `android.jar` for your API level:

```bash
./cli/hiddenjar build --api <level>          # macOS/Linux
# Windows: powershell -ExecutionPolicy Bypass -File cli\hiddenjar.ps1 build --api <level>
# any OS:  ./gradlew buildHiddenJar -Papi=<level>
```

Reopen if you still reproduce this on current tooling.

**Comment by KaelLuo** [comment]
您收到此邮件表示您发送的邮件我已经收到，会尽快给您回复，请耐心等候，亲
